feat: 독서록 조회 기능 컨트롤러 로직 작성

dani820 opened the pull request:
## 이 PR을 통해 해결하려는 문제
- 독서록 단 건 조회를 위한 조회 기능 컨트롤러 로직 작성

## 추가 및 변경사항
- 독서록 조회 기능 컨트롤러 로직 작성
- 테스트 코드 작성

## 참고(옵션)
- 테이블 설계 변경에 따른 Member 클래스 필드 정리 필요

## 체크리스트
- [x] 리뷰어를 지정하였는가
- [x] 코드가 오류나 경고없이 빌드되는가
- [x] 추가 및 변경된 사항에 대해 충분히 테스트 하였는가

dani820 committed:
feat: 독서록 조회 기능 컨트롤러 로직 작성

컨트롤러 로직 작성 및 테스트 코드 작성

f-lab-bradley commented on src/main/java/com/inmybook/adapter/in/web/dto/response/ContentResponse.java:
```diff
@@ -0,0 +1,13 @@
+public record ContentResponse(
+	String title,
+	String content,
+	String fromDate,
```
fromDate, toDate 가 무슨 뜻인가요???

dani820 replied:
독서 기간에 대한 변수로 각각 시작일, 종료일을 뜻하는데요. startDate, endDate로 변경하는 쪽이 더 의미가 맞다고 생각되어서 수정하려 합니다.

f-lab-bradley replied:
변수명이 명확하지 않아요.

f-lab-bradley commented on src/main/java/com/inmybook/adapter/in/web/dto/PostMapper.java:
```diff
@@ -3,7 +3,11 @@
+import com.inmybook.application.service.PostDetailsOutput;
 
 @Component
 public class PostMapper {
```
Mapper 는 왜 component 로 주입 하나요???

dani820 replied:
작성 당시 단순히 빈 등록을 위해 @Component를 사용했던 것 같습니다; 주입 방식을 달리 하는게 좋을까요?

f-lab-bradley replied:
static 함수로 만드는 방법도 고려해 보시면 좋을것 같아요.

dani820 committed:
refactor: 독서 시작일, 종료일 변수명 변경

fromDate -> readingStartDate
toDate -> readingEndDate

dani820 committed:
refactor: PostMapper 변수 static 으로 변경

f-lab-bradley commented on src/main/java/com/inmybook/adapter/in/web/PostController.java:
```diff
@@ -31,7 +31,7 @@ public class PostController {
 	private final RegisterPostUseCase registerPostUseCase;
 	private final ReadPostUseCase readPostUseCase;
-	private final PostMapper postMapper;
+	private static final PostMapper postMapper = new PostMapper();
```
객체 자체를 스태틱으로 선언 하는것 외에도 PostMapper 에서 제공하는 함수를 static 함수로 만드는걸 고려해 봐도 좋을 것 같아요.
객체로 따로 만들 필요가 없어 보이는 클래스처럼 보이긴 해서요

dani820 replied:
static 함수로 변경했습니다~

dani820 committed:
refactor: PostMapper 클래스 내 함수 static으로 변경

- PostMapper 클래스 내 함수들을 static으로 변경하여 불필요한 객체 생성 없이 간단히 접근 가능하도록 수정
- 조회 기능 Controller 테스트 코드 내에서 PostMapper 함수 모킹한 부분 수정(mockStatic 사용)

f-lab-bradley reviewed (commented):
구현 완료 된건가요??

dani820 commented:
넵 피드백 주신 부분 반영 완료했습니다

dani820 committed:
refactor: Member 관련 클래스 내 memberNo 필드 삭제

f-lab-bradley reviewed (approved):
수고 하셨습니당
